x/utils/fn/fn.go

```diff
@@ -0,0 +1,307 @@
+package fn
```

Comment:
I don't believe this package should live under the /x/ folder since that's cosmos-sdk nomenclature for cosmos-sdk modules, which this is not

Also none of these functions have tests?

also can we call the package allorautils or something, `fn` is such a short non descriptive name

also maybe this and the migutils should go in a separate PR so this PR is more tightly scoped to just a change about non-deferred close checking

Reply:
removed, but really would like to keep `migutils` in here since the code didn't actually change at all -- it's just ported verbatim to reduce duplication